virtual_ship/instruments/argo.py

```diff
@@ -0,0 +1,168 @@
+"""Argo instrument."""
+
+import math
+from dataclasses import dataclass
+from datetime import timedelta
+
+import numpy as np
+from parcels import (
+    AdvectionRK4,
+    FieldSet,
+    JITParticle,
+    ParticleSet,
+    StatusCode,
+    Variable,
+)
+
+from .location import Location
+
+
+@dataclass
+class Argo:
+    """Configuration for a single Argo."""
+
+    location: Location
+    deployment_time: float
+    min_depth: float
+    max_depth: float
+    drift_depth: float
+    vertical_speed: float
+    cycle_days: float
+    drift_days: float
+
+
+class _ArgoParticle(JITParticle):
+    cycle_phase = Variable("cycle_phase", dtype=np.int32, initial=0.0)
+    cycle_age = Variable("cycle_age", dtype=np.float32, initial=0.0)
+    drift_age = Variable("drift_age", dtype=np.float32, initial=0.0)
+    salinity = Variable("salinity", initial=np.nan)
+    temperature = Variable("temperature", initial=np.nan)
+    min_depth = Variable("min_depth", dtype=np.float32)
+    max_depth = Variable("max_depth", dtype=np.float32)
+    drift_depth = Variable("drift_depth", dtype=np.float32)
+    vertical_speed = Variable("vertical_speed", dtype=np.float32)
+    cycle_days = Variable("cycle_days", dtype=np.int32)
+    drift_days = Variable("drift_days", dtype=np.int32)
```

Comment:
I am not sure if this is cleaner. If we later introduce a type checker this is probably harder to check than the original class.

Reply:
It's cleaner because users can easily create bugs/issues when the varname on the left side of the equal sign is not the same as the varname in the Variable object; that is why I want to move away from the `varname = Variable(varname)` way of declaring. So I would strongly advice to change to the new way of creating a new Particle Class

Reply:
Removing duplication makes sense, I replaced. Regarding parcels, you could also look at the dataclass and fields API and try do something similar: https://www.studytonight.com/post/python-dataclass-field-function-part-3